matcher_test.go

```diff
@@ -0,0 +1,84 @@
+//go:build linux
+
+package goss
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/goss-org/goss/util"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	update = flag.Bool("update", false, "update the golden files of this test")
```

Comment:
Mind adding a comment for how and when to use this?

Reply:
Added comment